src/System.Private.CoreLib/src/System/Threading/ThreadPoolBoundHandle.PlatformNotSupported.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.IO;
+
+namespace System.Threading
+{
+    public sealed class ThreadPoolBoundHandle : IDisposable
+    {
```

Comment:
This should have explicit dummy private constructor. Otherwise, the C# compiler will generate a default public constructor.